File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    cnt = [0] * (number + 1)
    
    for i in range(1, number + 1):
        for j in range(i, number + 1, i):
            cnt[j] += 1
            
        if cnt[i] > limit:
            answer += power
        else:
            answer += cnt[i]

    return answer

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int number, int limit, int power) {
    int answer = 0;
    vector<int> check(number + 1) ;
    
    for(int i = 1 ; i <= number ; i++)
    {
        for(int j = i ; j <= number ; j+=i)
        {
            check[j]++ ;
        }
        check[i] > limit ? answer += power : answer += check[i] ;
    }
    
    return answer;
}